### DISCUSSION TITLE
Problems getting prices for tomorrow

### DISCUSSION THREAD

**Discussion by rstcologne** [BeatsuDev/tibber.py#45] [General]
Hey, 

I'm trying to collect the next day price data. Getting todays prices works for me like this:

account = tibber.Account(token) 
home = account.homes[0]  
subscription = home.subscriptions[0]  
prices = subscription.price_info.**today**

I figured, I should be able to get tomorrows prices when changing the last line to:

prices = subscription.price_info.**tomorrow**

However I always get an empty set even after 1 o'clock when the next day prices are available in the API explorer. 

What am I missing?

Thanks a lot,

Richard

**Comment by ericbstie** [comment]
Hi, thank you for bringing this up! I believe this is a bug with the Tibber API, since the raw data response from the API says that it's simply not receiving any data for the tomorrow field for the subscriptions (tested on the demo account). I noticed that if you get the subscription with the `current_subscription` property on the Home object, then it works though.

Recreating it in tibber.py:
```py
import tibber

home = tibber.Account(tibber.DEMO_TOKEN).homes[0]

# Does not work
home.subscriptions[0].price_info.tomorrow # [] (empty)
home.current_subscription.price_info.tomorrow # [Price(...), Price(...), ...]
```
Recreating it in the tibber api explorer:
```
{
  viewer {
    homes {
      currentSubscription {
        status
        priceInfo {
          tomorrow {total}
        }
      }
      subscriptions {
        status
        priceInfo {
          tomorrow {total}
        }
      }
    }
  }
}

```

So for now the solution would be to use home.current_subscription. Let me know if this does not fix your problem and if this bug only appears on the demo account. If this does not solve the issue, create a GitHub issue for this as this is probably a bug, then I'll get to it as soon as I have time.

**Reply by rstcologne** [reply]
Wow, that was quick. Thanks a lot! I'll give it a shot.

**Reply by rstcologne** [reply]
One more update. Aparently it was an issue on the server/API side. After writing my request, I experimented briefly with another library. Went back to the code with yours and now it works without implementing the suggestion you gave above.

**Reply by ericbstie** [reply]
Nice! :smile:
